### PR TITLE
Count linear extensions using hpolytope

### PR DESCRIPTION
**Pull request by vaithak**
This pull request provides an example for (approximately) counting the number of linear extensions of a post using volume computation of the order polytope (currently represented as an instance of hpolytope).

### CODE REVIEW

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+typedef typename HPOLYTOPE::VT VT;
+
+struct ArgOptions {
+	VOL_OPTIONS vo;
+	ROUND_OPTIONS ro;
+	bool with_rounding;
+	HPOLYTOPE* HP;
+
+	NT volume_method (HPOLYTOPE& P, 
+					  NT e,
```
Comment (vissarion) [inline examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: too many spaces here

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+typedef double NT;
+typedef Cartesian <NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef BoostRandomNumberGenerator<boost::mt19937, NT, 5> RNGType;
+typedef HPolytope <Point> HPOLYTOPE;
+typedef typename HPOLYTOPE::MT MT;
+typedef typename HPOLYTOPE::VT VT;
+
+struct ArgOptions {
+	VOL_OPTIONS vo;
```
Comment (vissarion) [inline examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: please use a 4 spaces indentation as in other files of the library

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+
+*/
+int main(int argc, char* argv[]) {
+	ArgOptions args;
+	if (!parseArgs(argc, argv, args)) {
+		std::cout<<"Errors occured";
+		return 0;
+	}       
+
+	std::cout<<calculateLinearExtension(args)<<"\n";
```
Comment (vissarion) [inline examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: please use spaces around operators i.e. `<<`

File: examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+0 0 0 0 1 0 1 0
```
Comment (vissarion) [inline examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt:1]: could you add the source of the file and how you generated in a comment?

**Review by vissarion — COMMENTED** [review]
Thanks! I am ok with merging after fixing some minor issues.

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp (hunk 2 above)
Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: Done :+1:

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp (hunk 3 above)
Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: Added spaces :+1:

File: examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt (hunk 4 above)
Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt:1]: I generated it using the instances given in the [le-counting-practice](https://github.com/vaithak/le-counting-practice) repo. Should I add the details in README file or a comment in C++ source code?

Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt:1]: Also I was thinking of adding the plots of corresponding DAGs of these instances in the README file.

File: examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp (hunk 1 above)
Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/volesti_lecount.cpp]: Resolved :+1:

File: examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt (hunk 4 above)
Comment (vissarion) [reply examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt:1]: >I generated it using the instances given in the le-counting-practice repo. Should I add the details in README file or a comment in C++ source code?

README should be fine

Comment (vaithak) [reply examples/count-linear-extensions-using-hpolytope/instances/bipartite_0.5_008_0.txt:1]: Added info regarding sample instances in the README file.

**Review by vissarion — APPROVED** [review]

**Comment by TolisChal** [comment]
@vaithak Thank you very much for this PR!
These are nice examples.

**Review by TolisChal — APPROVED** [review]

**Pull request merged**